src/main/resources/META-INF/resources/primefaces/datatable/datatable.js

```diff
@@ -1560,14 +1560,46 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
      */
     setOuterWidth: function(element, width) {
         if (element.css('box-sizing') === 'border-box') { // Github issue: #5014
-            var diff = element.outerWidth() - element.width();
-            element.width(parseFloat(width) - diff);
+            element.outerWidth(width);
         }
         else {
             element.width(width);
         }
     },
 
+    getColumnWidthInfo: function(col, isIncludeResizeableState) {
```

Comment:
Can you add JSDOC to this method and test that the build works with `mvn clean package -Pjsdoc` please?

Reply:
Done